src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -880,6 +880,13 @@ def visit_Assign(self, node: ast.Assign) -> list:
 
         return result
 
+    def visit_AugAssign(self, node: ast.AugAssign):
+        """Implement left <op>= right in terms of left = left <op> right."""
+        binary_operation = ast.BinOp(left=node.target, op=node.op, right=node.value)
+        assignment = ast.Assign(targets=[node.target], value=node.target)
+        ast.copy_location(assignment, binary_operation)
```

Comment:
This should propagate the location (according to the docs). I still have to put in a test for this.

Reply:
I wasn't aware of this helper function. That's quite useful, but I think this is still not quite right. Only the AugAssign node has a location so `copy_location` should be applied to the two new nodes `assignment` and `binary_operation`, i.e.
```
ast.copy_location(node, binary_operation)
ast.copy_location(node, assignment)
```

Reply:
That's a good point. But the column offset on the binary_operation will then be wrong. I will check if there is a simple way to correct it.

Reply:
Why is that? I would have assumed the location to point to the AugAssignment, i.e. 
```
a += b
   ^
```
Which I assume is the right location for both the assignment and the binary operation

Reply:
This implements the AugAssignment in terms of Assignment:

```
a += b  --> a = a + b
  ^(1)        ^(2)^(3)
```
 
`(1)` Is the column offset of the ast.AugAssign node, `(2)` of the new ast.Assign node, (3) of the new ast.BinOp node.
`(1) == (2)` but `(1) != (3)`. For a first approximation we can leave it like this now, especially since it has no immediate effect due to #122.

Btw, copy_location takes the new node first:
```
Signature: ast.copy_location(new_node, old_node)
```

For now it is impossible to even test any of this, since the source location information is lost later on anyway: #122